qiskit/synthesis/linear/linear_matrix_utils.py

```diff
@@ -12,164 +12,16 @@
 
 """Utility functions for handling binary matrices."""
 
-from typing import Optional, Union
-import numpy as np
-from qiskit.exceptions import QiskitError
-
-
-def check_invertible_binary_matrix(mat: np.ndarray):
```

Comment:
These functions used to accept also integer matrices, but now strictly require boolean types. Since these were not marked as private I'm not sure if we can change this -- maybe yes because they didn't appear in the API docs as far as I can tell 🤔 alternatively we could keep the Python function and normalize the input before using the Rust function, like
```python
def check_invertible_binary_matrix(mat):
    return _rust_implementation(mat.astype(bool))
```

Reply:
by their name and also the docstring below, these functions are for Boolean matrices

Reply:
I'm good with the stricter input type, I was just concerned that they might be public -- but if they're not I'm good with the change 👍🏻